src/meili/traffic_segment_matcher.cc

```diff
@@ -122,11 +126,15 @@ namespace {
           merged.back()->end_percent_ = segment.end_percent_;
           merged.back()->ends_segment_ = segment.ends_segment_;
           merged.back().internal = merged.back().internal && is_internal(directed_edge);
+          // Turn off turn channel flag if any segment continuation is not a turn channel
+          if (merged.back().turn_channel && !is_turn_channel(directed_edge))
```

Comment:
instead of a branch cant you just do: `merged.back().turn_channel = merged.back().turn_channel && is_turn_channel(directed_edge)`